homeassistant/components/thermopro/button.py

```diff
@@ -0,0 +1,169 @@
+"""Demo platform that offers a fake button entity."""
+
+from __future__ import annotations
+
+import logging
+
+from thermopro_ble import SensorUpdate, ThermoProBluetoothDeviceData, ThermoProDevice
+
+from homeassistant.components.bluetooth import (
+    BluetoothServiceInfoBleak,
+    async_ble_device_from_address,
+    async_track_unavailable,
+)
+from homeassistant.components.button import ButtonEntity, ButtonEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.dispatcher import (
+    async_dispatcher_connect,
+    async_dispatcher_send,
+)
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+from homeassistant.util.dt import now
+
+from .const import DOMAIN, SIGNAL_AVAILABILITY_UPDATED, SIGNAL_DATA_UPDATED
+
+_LOGGER = logging.getLogger(__name__)
+
+DATETIME_UPDATE = ButtonEntityDescription(
+    key="datetime",
+    translation_key="set_datetime",
+    icon="mdi:calendar-clock",
+    entity_category=EntityCategory.CONFIG,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up the demo button platform."""
+    assert entry.unique_id is not None
+
+    set_datetime_entity = None
+
+    @callback
+    def _async_on_data_updated(
+        data: ThermoProBluetoothDeviceData,
+        service_info: BluetoothServiceInfoBleak,
+        update: SensorUpdate,
+    ) -> None:
+        nonlocal set_datetime_entity
+        _LOGGER.debug(
+            "got update data=%s update=%s service_info=%s", data, update, service_info
+        )
+
+        assert entry.unique_id is not None
+        assert None in update.devices
+        assert update.devices[None].model is not None
+
+        if update.devices[None].model not in ("TP358", "TP393"):
+            return
+
+        if not set_datetime_entity:
+            name = update.devices[None].name
+            assert name is not None
+            set_datetime_entity = ThermoProDateTimeButtonEntity(
+                entry_id=entry.entry_id,
+                address=entry.unique_id,
+                device_name=name,
+                description=DATETIME_UPDATE,
+            )
+
+            async_add_entities(
+                [
+                    set_datetime_entity,
+                ]
+            )
+
+        if service_info.connectable and not set_datetime_entity.available:
+            _LOGGER.debug("sending availability '%s' for %s", True, entry.entry_id)
+            async_dispatcher_send(
+                hass,
+                f"{SIGNAL_AVAILABILITY_UPDATED}_{entry.entry_id}_button_datetime",
```

Comment:
```suggestion
                f"{SIGNAL_AVAILABILITY_UPDATED}_{entry.entry_id}",
```

We can leave off `button_datetime` since its watching for the address